packages/cli/src/typedSqlTagTransformer.ts

```diff
@@ -0,0 +1,165 @@
+    private readonly transform: TSTypedSQLTagTransformConfig,
+  ) {
+    this.includePattern = `${this.config.srcDir}/**/${transform.include}`;
+    this.localFileName = `${this.config.srcDir}${this.transform.emitFileName}`;
+    this.fullFileName = path.relative(process.cwd(), this.localFileName);
+  }
+
+  private async watch() {
+    let initialized = false;
+
+    const cb = async (fileName: string) => {
+      const job = {
+        files: [fileName],
+      };
+      !initialized
+        ? this.pushToQueue(job)
+        : await this.generateTypedSQLTagFileForJob(job, true);
+    };
+
+    chokidar
+      .watch(this.includePattern, {
+        persistent: true,
+        ignored: [this.localFileName],
+      })
+      .on('add', cb)
+      .on('change', cb)
+      .on('unlink', async (file) => await this.removeFileFromCache(file))
+      .on('ready', async () => {
+        initialized = true;
+        await this.waitForTypedSQLQueueAndGenerate(true);
+      });
+  }
+
+  public async start(watch: boolean) {
+    if (watch) {
+      return this.watch();
+    }
+
+    const fileList = globSync(this.includePattern, {
+      ignore: [this.localFileName],
+    });
+
+    debug('found query files %o', fileList);
+
+    await this.generateTypedSQLTagFileForJob({
+      files: fileList,
+    });
+  }
+
+  private pushToQueue(job: TransformJob) {
+    this.workQueue.push(
+      ...job.files.map((fileName) => this.getTsTypeDecs(fileName)),
+    );
+  }
+
+  private async getTsTypeDecs(
+    fileName: string,
+  ): Promise<TypedSQLTagTransformResult> {
+    console.log(`Processing ${fileName}`);
+    return (await this.pool.run(
+      {
+        fileName,
+        transform: this.transform,
+      },
+      'getTypeDecs',
+    )) as Awaited<getTypeDecsFnResult>;
+    // Result should be serializable!
+  }
+
+  private async generateTypedSQLTagFileForJob(
+    job: TransformJob,
+    useCache?: boolean,
+  ) {
+    this.pushToQueue(job);
+    return this.waitForTypedSQLQueueAndGenerate(useCache);
+  }
+
+  private async waitForTypedSQLQueueAndGenerate(useCache?: boolean) {
+    const queueResults = await Promise.all(this.workQueue);
+    this.workQueue.length = 0;
+
+    const typeDecsSets: TypeDeclarationSet[] = [];
+
+    for (const result of queueResults) {
+      if (result?.typedQueries.length) {
+        typeDecsSets.push(result);
+        if (useCache) this.cache[result.fileName] = result;
+      }
+    }
+
+    return this.generateTypedSQLTagFile(
+      useCache ? Object.values(this.cache) : typeDecsSets,
+    );
+  }
+
+  private async removeFileFromCache(fileToRemove: string) {
+    delete this.cache[fileToRemove];
+    return this.generateTypedSQLTagFile(Object.values(this.cache));
+  }
+
+  private contentStart = `/* eslint-disable */\nimport { ${this.transform.functionName} as sourceSql } from '@pgtyped/runtime';\n\n`;
+  private contentEnd = [
+    `export function ${this.transform.functionName}(s: string): unknown;`,
```

Comment:
When testing locally; I found I had to remove this line to get type-completion to work properly